### PR TITLE
[network] remove class NetworkAccessPoint and unused dialog CGUIDialogAccessPoints

### PR DESCRIPTION
**Pull request by lrusak**
This is legacy and unused code. Let's just get rid of it :rocket:

### CODE REVIEW

**Review by Rechi — CHANGES_REQUESTED** [review]
`CNetworkInterface::GetName`, `CNetworkInterface::IsWireless` and `CNetworkBase::GetInterfaceByName`  are now also unused.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
@Rechi better now?

File: xbmc/platform/win10/network/NetworkWin10.h [hunk 1]
```diff
@@ -39,17 +36,6 @@ class CNetworkInterfaceWin10 : public CNetworkInterface
...
-                         , std::string& essId, std::string& key, EncMode& encryptionMode) const;
-  virtual void SetSettings(const NetworkAssignment& assignment, const std::string& ipAddress
-                         , const std::string& networkMask, const std::string& defaultGateway
-                         , const std::string& essId, const std::string& key, const EncMode& encryptionMode);
-
-  // Returns the list of access points in the area
-  virtual std::vector<NetworkAccessPoint> GetAccessPoints(void) const;
 
 private:
   std::string m_adaptername;
```
Comment (Rechi) [inline xbmc/platform/win10/network/NetworkWin10.h]: only assigned, but value is never used

File: xbmc/platform/win32/network/NetworkWin32.h [hunk 2]
```diff
@@ -25,12 +25,9 @@ class CNetworkInterfaceWin32 : public CNetworkInterface
    CNetworkInterfaceWin32(const IP_ADAPTER_ADDRESSES& adapter);
    ~CNetworkInterfaceWin32(void) override;
 
-   const std::string& GetName(void) const override;
-
    bool IsEnabled(void) const override;
    bool IsConnected(void) const override;
-   bool IsWireless(void) const;
-
+
```
Comment (Rechi) [inline xbmc/platform/win32/network/NetworkWin32.h]: unrelated space change

**Review by Rechi — COMMENTED** [review]
Better, but `CNetworkInterfaceWin10::m_adaptername`, `CNetworkInterfaceWin10::m_winRT`, `CNetworkInterfaceWin10::m_profile`, `CNetworkInterfaceWin32::m_adaptername` and some includes are obsolete too.

**Comment by lrusak** [comment]
> Better, but `CNetworkInterfaceWin10::m_adaptername`, `CNetworkInterfaceWin10::m_winRT`, `CNetworkInterfaceWin10::m_profile`, `CNetworkInterfaceWin32::m_adaptername` and some includes are obsolete too.

I can try but I would only be poking in the dark and would be relying on jenkins. I could use some help here.

**Review by Rechi — APPROVED** [review]

**Review by notspiff — APPROVED** [review]

**Pull request merged**